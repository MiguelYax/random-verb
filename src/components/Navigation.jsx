import React from 'react'
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faRandom, faAdjust } from '@fortawesome/free-solid-svg-icons'
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Navigation = ({ darkMode, setDarkMode }) => {
    const navClassNames = classnames({
        "dark": darkMode,
        "light": !darkMode,
        'text-light': darkMode, 
        'text-primary': !darkMode
    });

    const buttonClassNames = classnames({
        "primary": !darkMode,
        "light": darkMode,
    })

    const textClassNames = classnames( {
        'text-light': darkMode, 
        'text-primary': !darkMode
    });
    
    return (
        <Navbar bg={navClassNames} expand="xl">
            <Navbar.Brand className={textClassNames} href="#home">
                <FontAwesomeIcon className="mr-2" icon={faRandom} />
                Random Verb
            </Navbar.Brand>
            <Nav>
                <Form inline>
                    <Button id="darkModeButton" variant={buttonClassNames} onClick={() => {
                        setDarkMode(prev => !prev);
                    }}>
                        <FontAwesomeIcon icon={faAdjust} />
                    </Button>
                </Form>
            </Nav>
        </Navbar>
    )
}

Navigation.propTypes = {
    darkMode: PropTypes.bool,
    setDarkMode: PropTypes.func,
}

export default Navigation;
