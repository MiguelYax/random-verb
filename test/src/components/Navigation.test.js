import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Navigation from '../../../src/components/Navigation'

describe('when component src/components/Navigation.jsx is used', () => {
    describe('when Navigation called with darkMode equal true', () => {
        it('shold be render the dark mode ', () => {
            const app = renderer.create(<Navigation darkMode={true} setDarkMode={() => {}} />).toJSON();
            expect(app).toMatchSnapshot();
        });
    });
    describe('when Navigation called with darkMode equal false', () => {
        it('shold be render the light mode ', () => {
            const app = renderer.create(<Navigation darkMode={false} setDarkMode={() => {}} />).toJSON();
            expect(app).toMatchSnapshot();
        });
    });
    describe('when darkMode button is cliked', () => {
        const setDarkMode = jest.fn();
        const NavigationMounted = mount((<Navigation darkMode={false} setDarkMode={setDarkMode} />));
        const darkModeButton = NavigationMounted.find('#darkModeButton');

        expect(darkModeButton.length).toBe(1);
        darkModeButton.simulate('click')
        expect(setDarkMode.mock.calls.length).toBe(1);
    });
});
