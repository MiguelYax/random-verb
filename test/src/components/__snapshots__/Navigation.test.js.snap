// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when component src/components/Navigation.jsx is used when Navigation called with darkMode equal false should be render the light mode  1`] = `
<Navbar
  bg="light text-primary"
  collapseOnSelect={false}
  expand="xl"
  variant="light"
>
  <NavbarBrand
    className="text-primary"
    href="#home"
  >
    <FontAwesomeIcon
      border={false}
      className="mr-2"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f074",
            "M504.971 359.029c9.373 9.373 9.373 24.569 0 33.941l-80 79.984c-15.01 15.01-40.971 4.49-40.971-16.971V416h-58.785a12.004 12.004 0 0 1-8.773-3.812l-70.556-75.596 53.333-57.143L352 336h32v-39.981c0-21.438 25.943-31.998 40.971-16.971l80 79.981zM12 176h84l52.781 56.551 53.333-57.143-70.556-75.596A11.999 11.999 0 0 0 122.785 96H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12zm372 0v39.984c0 21.46 25.961 31.98 40.971 16.971l80-79.984c9.373-9.373 9.373-24.569 0-33.941l-80-79.981C409.943 24.021 384 34.582 384 56.019V96h-58.785a12.004 12.004 0 0 0-8.773 3.812L96 336H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h110.785c3.326 0 6.503-1.381 8.773-3.812L352 176h32z",
          ],
          "iconName": "random",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Random Verb
  </NavbarBrand>
  <Nav
    fill={false}
    justify={false}
  >
    <Form
      inline={true}
    >
      <Button
        active={false}
        disabled={false}
        id="darkModeButton"
        onClick={[Function]}
        variant="primary"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f042",
                "M8 256c0 136.966 111.033 248 248 248s248-111.034 248-248S392.966 8 256 8 8 119.033 8 256zm248 184V72c101.705 0 184 82.311 184 184 0 101.705-82.311 184-184 184z",
              ],
              "iconName": "adjust",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Button>
    </Form>
  </Nav>
</Navbar>
`;

exports[`when component src/components/Navigation.jsx is used when Navigation called with darkMode equal true shold be render the dark mode  1`] = `
<Navbar
  bg="dark text-light"
  collapseOnSelect={false}
  expand="xl"
  variant="light"
>
  <NavbarBrand
    className="text-light"
    href="#home"
  >
    <FontAwesomeIcon
      border={false}
      className="mr-2"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f074",
            "M504.971 359.029c9.373 9.373 9.373 24.569 0 33.941l-80 79.984c-15.01 15.01-40.971 4.49-40.971-16.971V416h-58.785a12.004 12.004 0 0 1-8.773-3.812l-70.556-75.596 53.333-57.143L352 336h32v-39.981c0-21.438 25.943-31.998 40.971-16.971l80 79.981zM12 176h84l52.781 56.551 53.333-57.143-70.556-75.596A11.999 11.999 0 0 0 122.785 96H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12zm372 0v39.984c0 21.46 25.961 31.98 40.971 16.971l80-79.984c9.373-9.373 9.373-24.569 0-33.941l-80-79.981C409.943 24.021 384 34.582 384 56.019V96h-58.785a12.004 12.004 0 0 0-8.773 3.812L96 336H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h110.785c3.326 0 6.503-1.381 8.773-3.812L352 176h32z",
          ],
          "iconName": "random",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Random Verb
  </NavbarBrand>
  <Nav
    fill={false}
    justify={false}
  >
    <Form
      inline={true}
    >
      <Button
        active={false}
        disabled={false}
        id="darkModeButton"
        onClick={[Function]}
        variant="light"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f042",
                "M8 256c0 136.966 111.033 248 248 248s248-111.034 248-248S392.966 8 256 8 8 119.033 8 256zm248 184V72c101.705 0 184 82.311 184 184 0 101.705-82.311 184-184 184z",
              ],
              "iconName": "adjust",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Button>
    </Form>
  </Nav>
</Navbar>
`;
